isis/src/base/apps/noproj/noproj.cpp

```diff
@@ -184,11 +184,13 @@ namespace Isis {
 
     // Can we do a regular label? Didn't work on 12-15-2006
     cao.setLabelAttachment(Isis::DetachedLabel);
+    FileName matchCubeFile = FileName::createTempFile("./match.cub");
```

Comment:
This should be created in the users IsisPreference temp dir, not CWD. CWD could be a readonly file system.  Should be able to replace "./match.cub" with "$Temporary/match.cub". The expand will do the preference look up.

Reply:
This is an issue we also ran into when running noproj in the same directory. Stuart is right on about using the temporary name generation method in FileName and temporary directory.

The other problem we encountered was that some object files remained after processing. This is because noproj is very HiRISE specific in its cleanup. Here is some new code that shows how this can be generically handled using the match cube label:

```
//  Cleanup by deleting the match files
  QStringList detfiles = findAllDetachedFiles( label );
  detfiles.append(matchLbl);

  // Now actually remove the files
  foreach (const QString &dfile, detfiles ) {
    std::string  dtf = dfile.toStdString();
    remove ( dtf.c_str() );
  }
```

Here is the actual function referenced that uses the match file label to determine the detached object files that need to be deleted:

```

// Find all detached filenames specified in objects in the label
QStringList findAllDetachedFiles(const PvlObject &object) {

  // Check this object for a detached file spec
  QStringList detfiles;
  QString dfilename = "^" + object.name();
  if ( object.hasKeyword(dfilename) ) {
    QString detname = object[dfilename];
    detfiles.append(detname);
  }

  // Now check all objects contain in this object
  for (int i_obj = 0; i_obj <  object.objects(); i_obj++) {
    const PvlObject &obj = object.object(i_obj);
    QStringList files = findAllDetachedFiles(obj);
    if ( files.size() > 0 ) {
      detfiles.append(files);
    }
    
  }
  
  return ( detfiles );

}
```